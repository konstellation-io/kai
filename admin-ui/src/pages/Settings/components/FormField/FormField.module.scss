@import '../../../../styles/colors';
@import '../../../../styles/mixins';

.formField {
  text-transform: capitalize;
  height: 140px;
  display: flex;
  flex-direction: row;
  align-items: center;
  margin-top: 30px;
  border: solid $bg-color-lighter;
  border-width: 2px 0;
  flex-shrink: 0;

  .label {
    @include font-body;
    color: font-color(light);

    margin: 0 1vw 0 16px !important;
  }

  svg {
    margin-left: 20px;
  }

  .input {
    width: auto;
    flex-grow: 1;
  }

  .button {
    margin-left: 2vw;
  }
}
