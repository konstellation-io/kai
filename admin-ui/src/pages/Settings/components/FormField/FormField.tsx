import React, { MouseEvent, ReactElement } from 'react';
import styles from './FormField.module.scss';
import TextInput from '../../../../components/Form/TextInput/TextInput';
import Button from '../../../../components/Button/Button';

type FormFieldProps = {
  fieldName: string;
  value?: string;
  error: string;
  onChange: Function;
  onSubmit: (e: MouseEvent<HTMLDivElement>) => void;
  icon: ReactElement;
};

function FormField({
  fieldName,
  value,
  error,
  onChange,
  onSubmit,
  icon
}: FormFieldProps) {
  return (
    <div className={styles.formField}>
      {icon}
      <p className={styles.label}>{fieldName} whitelist</p>
      <div className={styles.input}>
        <TextInput
          whiteColor
          label={fieldName}
          error={error}
          onChange={onChange}
          onEnterKeyPress={onSubmit}
          formValue={value}
        />
      </div>
      <div className={styles.button}>
        <Button
          label={`ADD ${fieldName.toUpperCase()}`}
          onClick={onSubmit}
          border
        />
      </div>
    </div>
  );
}

export default FormField;
