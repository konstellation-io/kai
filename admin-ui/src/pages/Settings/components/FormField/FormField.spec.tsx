import React from 'react';
import { shallow } from 'enzyme';
import FormField from './FormField';

describe('FormField component', () => {
  let wrapper: any;

  const errorMock = 'foo';
  const onChangeMock = jest.fn();
  const onSubmitMock = jest.fn();
  const fieldNameMock = 'bar';
  const valueMock = 'baz';

  beforeEach(() => {
    jest.resetAllMocks();
    wrapper = shallow(
      <FormField
        icon={<></>}
        error={errorMock}
        onChange={onChangeMock}
        onSubmit={onSubmitMock}
        fieldName={fieldNameMock}
        value={valueMock}
      />
    );
  });

  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
