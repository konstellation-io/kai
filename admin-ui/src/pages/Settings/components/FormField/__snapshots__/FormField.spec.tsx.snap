// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormField component should render without crashing 1`] = `
<div
  className="formField"
>
  <p
    className="label"
  >
    bar
     whitelist
  </p>
  <div
    className="input"
  >
    <TextInput
      error="foo"
      formValue="baz"
      label="bar"
      onChange={[MockFunction]}
      onEnterKeyPress={[MockFunction]}
      whiteColor={true}
    />
  </div>
  <div
    className="button"
  >
    <Button
      border={true}
      label="ADD BAR"
      onClick={[MockFunction]}
    />
  </div>
</div>
`;
